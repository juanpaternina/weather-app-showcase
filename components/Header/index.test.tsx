import { render, screen } from '@testing-library/react-native';
const { Header } = require('.');

jest.mock('expo-router', () => ({
  Link: ({ children }) => children,
}));

jest.mock('@expo/vector-icons/MaterialIcons', () => ({
  __esModule: true,
  default: ({ name, size, color }) => (
    <mock-icon
      data-testid={`material-icon-${name}`}
      size={size}
      color={color}
    />
  ),
}));

describe('Header', () => {
  const defaultProps = {
    cityName: 'London',
    countryName: 'UK',
  };

  test('It should render Header component', () => {
    const { toJSON } = render(<Header {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('It should render Header component with the city and country name', () => {
    render(<Header {...defaultProps} />);
    expect(screen.getByText('London, UK')).toBeTruthy();
  });

  test('It should render Search button', () => {
    render(<Header {...defaultProps} />);
    const searchIcon = screen.getByTestId('material-icon-search');
    expect(searchIcon).toBeTruthy();
  });
});
