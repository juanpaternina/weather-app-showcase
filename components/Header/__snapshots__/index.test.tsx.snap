// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header It should render Header component 1`] = `
<View>
  <Text
    style={
      [
        {
          "alignSelf": "center",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "color": "black",
          "fontFamily": "bold",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
          "width": 250,
        },
        {
          "marginBottom": 4,
        },
      ]
    }
  >
    London
    , 
    UK
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "position": "absolute",
        "right": 20,
        "top": 0,
      }
    }
    testID="material-icon-search"
  >
    <mock-icon
      color="black"
      data-testid="material-icon-search"
      size={32}
    />
  </View>
</View>
`;
